pypeit/spectrographs/ntt_efosc2.py

```diff
@@ -0,0 +1,336 @@
+            if meta_key == 'datasec':
+                datasec = '[%s:%s,:%s]' % (pscan_x, pscan_x+data_x, data_y)
+                return datasec
+            else:
+                oscansec = '[:%s,:%s]' % (pscan_x, data_y) # Actually two overscan regions, here I only dealing with the region on x-axis
+                return oscansec
+        
+        else:
+            msgs.error("Not ready for this compound meta")
+
+
+    def get_detector_par(self, hdu, det):
+        """
+        Return metadata for the selected detector.
+
+        Args:
+            hdu (`astropy.io.fits.HDUList`_):
+                The open fits file with the raw image of interest.
+            det (:obj:`int`):
+                1-indexed detector number.
+
+        Returns:
+            :class:`~pypeit.images.detector_container.DetectorContainer`:
+            Object with the detector metadata.
+        """
+        # Binning
+        binning = self.get_meta_value(self.get_headarr(hdu), 'binning')
+        
+        # Manual: https://www.eso.org/sci/facilities/lasilla/instruments/efosc/doc/manual/EFOSC2manual_v4.2.pdf
+        # Instrument paper: http://articles.adsabs.harvard.edu/pdf/1984Msngr..38....9B
+        detector_dict = dict(
+            binning         = binning,
+            det             = 1, # only one detector
+            dataext         = 0,
+            specaxis        = 0,
+            specflip        = False,
+            spatflip        = False,
+            platescale      = 0.005, # focal length is 200mm, unit radian/mm, manual 2.2
+            darkcurr        = 0.0,
+            saturation      = 65535, # Maual Table 8
+            nonlinear       = 0.80,
+            mincounts       = -1e10,
+            numamplifiers   = 1,
+            gain            = np.atleast_1d(0.91), # written in hgeader['HIERARCH ESO DET OUT1 GAIN']
+            ronoise         = np.atleast_1d(10.0), # manual page 108
+            datasec         = np.atleast_1d(self.get_meta_value(self.get_headarr(hdu), 'datasec')),
+            oscansec        = np.atleast_1d(self.get_meta_value(self.get_headarr(hdu), 'oscansec'))
+            #suffix          = '_Thor',
+        )
+        return detector_container.DetectorContainer(**detector_dict)
+
+    @classmethod
+    def default_pypeit_par(cls):
+        """
+        Return the default parameters to use for this instrument.
+        
+        Returns:
+            :class:`~pypeit.par.pypeitpar.PypeItPar`: Parameters required by
+            all of ``PypeIt`` methods.
+        """
+        par = super().default_pypeit_par()
+
+        # Always correct for flexure, starting with default parameters
+        par['flexure']['spec_method'] = 'boxcar'
+
+        # Median overscan
+        #   IF YOU CHANGE THIS, YOU WILL NEED TO DEAL WITH THE OVERSCAN GOING ALONG ROWS
+        for key in par['calibrations'].keys():
+            if 'frame' in key:
+                par['calibrations'][key]['process']['overscan_method'] = 'median'
```

Comment:
if we are not overscan subtracting, remove this

Reply:
Yes.